lexer/lexer.go

```diff
@@ -32,6 +33,32 @@ type Position struct {
 	Column int
 }
 
+func (pos Position) Add(lines int, cols int) Position {
```

Comment:
I would rather we didn't add this `Add` method. It seems a bit weird to "add" to a position, and it's only used in one place, where I think it'd actually be preferable to make the error position the name itself, rather than just past the name.